### PR TITLE
Suggest command on spelling mistakes

### PR DESCRIPTION
**Pull request by fgaz**
Patch from Francesco Ariis

---
Please include the following checklist in your PR:

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#conventions).
* [x] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#changelog).
* [x] The documentation has been updated, if necessary.

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

File: Cabal/src/Distribution/Simple/Command.hs [hunk 1]
```diff
@@ -513,6 +529,27 @@ commandsRun globalCommand commands args =
 
      where globalHelp = commandHelp globalCommand
 
+-- Levenshtein distance, from https://wiki.haskell.org/Edit_distance
+-- (Author: JeanPhilippeBernardy, Simple Permissive Licence)
+editDistance :: Eq a => [a] -> [a] -> Int
```
Comment (phadej) [inline Cabal/src/Distribution/Simple/Command.hs:534]: This is a bit sad as `cabal-install` already depends on `edit-distance` package. (But adding dependency to `Cabal` is hard).

Comment (fgaz) [reply Cabal/src/Distribution/Simple/Command.hs:534]: Yeah :-/

I can't think of a better solution though. cabal-install's option parsing is heavily tied to Cabal's

Comment (phadej) [reply Cabal/src/Distribution/Simple/Command.hs:534]: Yes. It's not a great design, but that what we have now.

**Review by emilypi — COMMENTED** [review]
I agree with @phadej here. Not ideal, but definitely a quality of life improvemnt 👍

**Review by emilypi — APPROVED** [review]

**Pull request merged**